### PR TITLE
Added lambda for getHash and minor changes in uploadFile lambda

### LINKED ISSUE #37: [BUG] Update timestamp generator
**Describe the bug**
The timestamp should be generated such that it is in `YYYY-MM-DD-HH-mm-ss` format if there is any single digit in the timestamp append a `0` before it. Like if the original timestamp is like `2021-4-21-2-50-8`  it should be converted to `2021-4-21-02-50-08`.

Area Where Bug Occurred
- [X] Lambda <@lambda_for_fileUpload>
- [ ] DynamodDb

**How to reproduce the error/bug**
Use `date-and -time` library refer [https://www.npmjs.com/package/date-and-time](https://www.npmjs.com/package/date-and-time)

**Additional context**
Storing timestamp according to the proposed way can help in getting the files sorted on the basis of timestamp.

**Before submitting an issue check the below**
- [X] I have synced all my node versions as mentioned in the project"
- [X] I am using the same version of npm as is the project

### PR DESCRIPTION
**Pull request by lakshya-20**
- Added lambda function to get hash corresponding to an URL
- Updated generate timestamp function in lambda_for_uploadFile

### CODE REVIEW

File: lambdas/url_access_patterns/lambda_for_getHash/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  };
+}
+
+exports.getHash = async (event) =>{
+    try{
+        var params={
+            TableName:"V-Transfer",
+            IndexName:"FIND_FILE_BY_URLID",
+            KeyConditionExpression: "#PK= :pk",
+            ExpressionAttributeNames:{
```
Comment (danishjamal104) [inline lambdas/url_access_patterns/lambda_for_getHash/index.js]: can we remove this and define key in place?

File: lambdas/url_access_patterns/lambda_for_getHash/index.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                "#PK": "GS1_PK",            
+            },
+            ExpressionAttributeValues:{
+                ':pk':`${event.path.url_id}`,                
+            },                                    
+        }        
+        const url_data = await dynamo.query(params).promise()        
+        const return_data=url_data.Items[0]? url_data.Items[0].hash:"";
```
Comment (danishjamal104) [inline lambdas/url_access_patterns/lambda_for_getHash/index.js]: where is the conditional check happening, like if the number of clicks is available or not, and if the URL is visible or not?

**Review by danishjamal104 — CHANGES_REQUESTED** [review]

File: lambdas/url_access_patterns/lambda_for_getHash/index.js (hunk 2 above)
Comment (lakshya-20) [reply lambdas/url_access_patterns/lambda_for_getHash/index.js]: Noted.

File: lambdas/file_access_patterns/lambda_for_fileUpload/index.js [hunk 3]
```diff
@@ -8,17 +8,20 @@ function response(statusCode, message) {
     body: JSON.stringify(message),
   };
 }
+function generateTimestamp() {
+  const ts = Date.now();
+  const pattern = DT.compile("YYYY-MM-DD-HH-mm-ss");
```
Comment (ayush0x00) [inline lambdas/file_access_patterns/lambda_for_fileUpload/index.js]: DT is not defined.
`const DT=require("date-and-time")` also do the npm install to include packages in package.json

Comment (lakshya-20) [reply lambdas/file_access_patterns/lambda_for_fileUpload/index.js]: Noted

File: lambdas/url_access_patterns/lambda_for_getHash/index.js (hunk 1 above)
Comment (lakshya-20) [reply lambdas/url_access_patterns/lambda_for_getHash/index.js]: Do you mean one external template file for all lambda functions?

Comment (danishjamal104) [reply lambdas/url_access_patterns/lambda_for_getHash/index.js]: no, I mean using `KeyConditionExpression: "PK= :pk",`, and removing the 
```
ExpressionAttributeNames:{
    "#PK": "GS1_PK",            
}
```

**Comment by danishjamal104** [comment]
@lakshya-20 , please squash the commits. Thanks :)

File: lambdas/url_access_patterns/lambda_for_getHash/index.js (hunk 1 above)
Comment (ayush0x00) [reply lambdas/url_access_patterns/lambda_for_getHash/index.js]: yup we can do that

**Pull request closed without merging**